luigi_pipeline/sex_related_checks.py

```diff
@@ -0,0 +1,568 @@
+        :return: True or Exception
+        """
+        if genome_version == "GRCh37":
+            contig_check_result = contig_check(
+                mt, GRCh37_STANDARD_CONTIGS, VARIANT_THRESHOLD
+            )
+        elif genome_version == "GRCh38":
+            contig_check_result = contig_check(
+                mt, GRCh38_STANDARD_CONTIGS, VARIANT_THRESHOLD
+            )
+
+        if bool(contig_check_result):
+            err_msg = ""
+            for k, v in contig_check_result.items():
+                err_msg += "{k}: {v}. ".format(k=k, v=", ".join(v))
+            raise SeqrValidationError(err_msg)
+
+        sample_type_stats = HailMatrixTableTask.sample_type_stats(mt, genome_version)
+
+        for name, stat in sample_type_stats.items():
+            logger.info(
+                "Table contains %i out of %i common %s variants."
+                % (stat["matched_count"], stat["total_count"], name)
+            )
+
+        has_coding = sample_type_stats["coding"]["match"]
+        has_noncoding = sample_type_stats["noncoding"]["match"]
+
+        if not has_coding and not has_noncoding:
+            # No common variants detected.
+            raise SeqrValidationError(
+                "Genome version validation error: dataset specified as {} but doesn't contain "
+                "the expected number of common this build's variants".format(
+                    genome_version
+                )
+            )
+        elif has_noncoding and not has_coding:
+            # Non coding only.
+            raise SeqrValidationError(
+                "Sample type validation error: Dataset contains noncoding variants but is missing common coding "
+                "variants for {}. Please verify that the dataset contains coding variants.".format(
+                    genome_version
+                )
+            )
+        elif has_coding and not has_noncoding:
+            # Only coding should be WES.
+            if sample_type != "WES":
+                raise SeqrValidationError(
+                    "Sample type validation error: dataset sample-type is specified as WGS but appears to be "
+                    "WES because it contains many common coding variants"
+                )
+        elif has_noncoding and has_coding:
+            # Both should be WGS.
+            if sample_type != "WGS":
+                raise SeqrValidationError(
+                    "Sample type validation error: dataset sample-type is specified as WES but appears to be "
+                    "WGS because it contains many common non-coding variants"
+                )
+        return True
+
+    def import_dataset(self):
+        """
+        Imports VCF to a MatrixTable.
+
+        If optional WES filters VCF is passed, import and join with VCF.
+        :return: hl.MatirxTable
+        """
+        # Import the VCFs from inputs. Set min partitions so that local pipeline execution takes advantage of all CPUs.
+        recode = {}
```

Comment:
This seems to have a bunch of copy-paste from the `HailMatrixTableTask` task, we really should have this task inherit from that. Thats would also give us cleaner access to some of the class helper functions